src/services/studio/studio.service.ts

```diff
@@ -125,33 +124,46 @@ class StudioService {
         });
     }
 
-    private deleteFileIfTypeScenario(packageZip: AdmZip, entry: IZipEntry): void {
-        if (entry.entryName.startsWith("nodes/") && entry.entryName.endsWith(".yml")) {
-            const node: NodeExportTransport = parse(entry.getData().toString());
-            if (node.type === "SCENARIO") {
-                packageZip.deleteFile(entry);
-            }
-        }
+    private deleteScenarioAssets(packageZip: AdmZip): void {
+        packageZip.getEntries().filter(entry => entry.entryName.startsWith("nodes/") && entry.entryName.endsWith(".yml"))
+            .forEach(entry => {
+                const node: NodeExportTransport = parse(entry.getData().toString());
+                if (node.type === "SCENARIO") {
+                    packageZip.deleteFile(entry);
+                }
+            });
     }
 
-    private fixConnectionVariablesIfRootNodeFile(packageZip: AdmZip, entry: IZipEntry, zipName: string, exportedVariables: VariableManifestTransport[]): void {
-        if (entry.name === "package.yml") {
-            const packageKeyAndVersion = zipName.replace(".zip", "").split("_");
-            const connectionVariablesByKey = this.getConnectionVariablesByKeyForPackage(packageKeyAndVersion[0], packageKeyAndVersion[1], exportedVariables);
+    private fixConnectionVariablesForRootNodeFiles(packageZip: AdmZip, zipName: string, exportedVariables: VariableManifestTransport[]): void {
+        const packageKeyAndVersion = this.getPackageKeyAndVersion(zipName);
 
-            if (connectionVariablesByKey.size) {
-                const exportedNode: NodeExportTransport = parse(entry.getData().toString());
-                const nodeContent: NodeSerializedContent = parse(exportedNode.serializedContent);
+        const connectionVariablesByKey = this.getConnectionVariablesByKeyForPackage(packageKeyAndVersion.packageKey, packageKeyAndVersion.version, exportedVariables);
 
-                nodeContent.variables = nodeContent.variables.map(variable => ({
-                    ...variable,
-                    metadata: variable.type === PackageManagerVariableType.CONNECTION ?
-                        connectionVariablesByKey.get(variable.key).metadata : variable.metadata
-                }));
+        if (connectionVariablesByKey.size) {
```

Comment:
I would prefer this to be a number comparison.
` if(connectionVariablesByKey.size !== 0) {`
Alternatively, this could also be an early return:
```
if(connectionVariablesByKey.size === 0) {
    return;
}
```